src/Bicep.Core/Semantics/Namespaces/SystemNamespaceType.cs

```diff
@@ -1806,6 +1806,23 @@ static IEnumerable<Decorator> GetBicepTemplateDecorators(IFeatureProvider featur
                     .Build();
                 }
 
+                if (featureProvider.OnlyIfNotExistsEnabled)
+                {
+                    yield return new DecoratorBuilder(LanguageConstants.OnlyIfNotExistsPropertyName)
+                    .WithDescription("Causes the resource deployment to be skipped if the resource already exists")
+                    .WithFlags(FunctionFlags.ResourceDecorator)// the decorator is constrained to resources
+                    .WithEvaluator((functionCall, decorated) =>
+                    {
+                        if (decorated is DeclaredResourceExpression declaredResourceExpression)
+                        {
+                            declaredResourceExpression.DecoratorConfig.Add("OnlyIfNotExists", new ArrayExpression(null, []));
+                            return declaredResourceExpression;
+                        }
+                        return decorated;
+                    })
```

Comment:
Is it possible to make this logic generic, and create a reusable method for it?

E.g. so this code could be updated to just:
```csharp
.WithEvaluator(AddDecoratorConfigToResource)
.Build();
```

Where `AddDecoratorConfigToResource` could be defined as:
```csharp
private static Expression AddDecoratorConfigToResource(FunctionCallExpression expression, Expression decorated)
{
    ...
}
```